src/mslice/util/mantid/algorithm_wrapper.py

```diff
@@ -95,3 +95,12 @@ def add_to_ads(workspaces):
     for workspace in workspaces:
         startid = (5 if workspace.name.startswith('__mat') else 2) if workspace.name.startswith('__') else 0
         AnalysisDataService.Instance().addOrReplace(workspace.name[startid:], workspace.raw_ws)
+
+
+def remove_from_ads(workspacename):
+    if workspacename in AnalysisDataService.Instance().getObjectNames():
+        AnalysisDataService.Instance().remove(workspacename)
+    # Remove hidden workspaces from ADS
+    workspacename = '__MSL' + workspacename
+    if workspacename in AnalysisDataService.Instance().getObjectNames():
```

Comment:
Interesting. When I create a few cut workspaces and delete them from MD Histo and then try to retrieve them either from the Python editor in Mantid or in MSlice I get a KeyError.

Reply:
Also, the deleted cut workspaces disappear from the workspace window in Mantid workbench.

Reply:
According to the documentation getObjectNames returns a list of all names currently known to the ADS. These hidden workspaces have been added to the ADS before.

Reply:
Oh thats intersting, maybe I have a different `DataServiceHidden::Auto` configuration to you?

https://github.com/mantidproject/mantid/blob/ab97a0be2875f9833b22233245f20eff051a44fa/Framework/Kernel/inc/MantidKernel/DataService.h#L393

Reply:
Ah, I think my `showingHiddenObjects()` returns false. So I guess this means the hidden cut workspaces will only be deleted if Show hidden workspaces in the Settings is set to true?

Reply:
Ah ok, `getObjectNames` checks whether the setting to include hidden objects is set. I have to ensure that this is set to `Include` as the setting within Mantid might be different for the user 🤦

Reply:
> Oh thats intersting, maybe I have a different `DataServiceHidden::Auto` configuration to you?
> 
> https://github.com/mantidproject/mantid/blob/ab97a0be2875f9833b22233245f20eff051a44fa/Framework/Kernel/inc/MantidKernel/DataService.h#L393

Yes, while working on this I have set this to true. That won't be the case for some of the users though, so it has to be set to true in the call of `getObjectNames`.

Reply:
I think enabling the "Show Invisible Workspaces" option in user settings temporarily is the most elegant solution to access the hidden workspaces in the ADS when calling `getObjectNames`. I immediately restore it to the original setting so the user should not experience any difference. 
The alternative would be a change to the Python interface for `getObjectNames` that does not seem to be required by any other interface.

Reply:
Do you know if `AnalysisDataService.Instance().doesExist` has the same behaviour as `getObjectNames`? i.e. maybe it could be used on line 115 to check if a workspace is in the ADS, and this might mean we won't need to change the configservice settings if it is independent from the "Show invisible workspaces" option?